identity/webapp/pages/api/registration-form.ts

```diff
@@ -0,0 +1,29 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { generateNewToken, decodeToken } from '../../src/utility/jwt-codec';
+
+export default (req: NextApiRequest, res: NextApiResponse): void => {
+  if (req.method === 'POST') {
+    const {
+      state,
+      redirectUri,
+      firstName,
+      lastName,
+      termsAndConditions,
+      sessionToken,
+    } = req.body;
+
+    const decodedToken = decodeToken(sessionToken);
+    const formData = { firstName, lastName, termsAndConditions };
+
+    if (typeof decodedToken !== 'string') {
+      const newToken = generateNewToken(decodedToken, state, formData);
+
+      res.redirect(
+        308,
+        `${redirectUri}?state=${state}&session_token=${newToken}`
```

Comment:
Need to consider whether passing an arbitrary `redirectUri` here could be harmful - elsewhere (callback and login URLs) we have to provide a whitelist. Not sure whether that's required here.

Reply:
That's a good point – we wouldn't expect the `redirectUri` to change would we?

Reply:
True, it's always going to be this `/continue` endpoint right? I think we should be able to construct that from the Auth0 config variables that we already have

Reply:
I think we can do as Jamie suggests and construct this from the Auth0 config variables